plugins/modules/registration_command.py

```diff
@@ -0,0 +1,204 @@
+    required: false
+    type: str
+  packages:
+    description:
+    - Packages to install on the host when registered.
+    required: false
+    type: str
+  remote_execution_interface:
+    description:
+    - Identifier of the Host interface for Remote execution
+    required: false
+    type: str
+  repo:
+    description:
+    - Repository URL (yum/dnf) or full sources.list entry (apt).
+    required: false
+    type: str
+  repo_gpg_key_url:
+    description:
+    - URL of the GPG key for the repository
+    required: false
+    type: str
+  setup_insights:
+    description:
+    - If this is set to C(true), insights client will be installed
+      and registered on Red Hat family operating systems.
+    required: false
+    type: bool
+  setup_remote_execution:
+    description:
+    - If this is set to true, SSH keys will be installed on the host.
+    required: false
+    type: bool
+  setup_remote_execution_pull:
+    description:
+    - If this is set to true, pull provider client will be deployed on the host.
+    required: false
+    type: bool
+  smart_proxy:
+    description:
+    - Name of Smart Proxy.
+    - This Proxy must have both the C(Templates) and C(Registration) features enabled.
+    required: false
+    type: str
+  update_packages:
+    description:
+    - Update all packages on the host
+    required: false
+    type: bool
+  organization:
+    description:
+      - Organization to register the host in
+    required: false
+    type: str
+  location:
+    description:
+      - Location to register the host in
+    required: false
+    type: str
+
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+'''
+
+EXAMPLES = '''
+- name: "Generate registration command"
+  theforeman.foreman.registration_command:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+  register: command
+
+- name: "Perform registration"
+  ansible.builtin.shell:
+    cmd: "{{ command.registration_command }}"
+'''
+
+RETURN = '''
+registration_command:
+  description: The generated registration command.
+  returned: success
+  type: str
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+class ForemanRegistrationCommandModule(ForemanAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanRegistrationCommandModule(
+        foreman_spec=dict(
+            hostgroup=dict(type='entity'),
+            operatingsystem=dict(type='entity'),
+            smart_proxy=dict(type='entity'),
+            setup_insights=dict(type='bool'),
+            setup_remote_execution=dict(type='bool'),
+            jwt_expiration=dict(type='int'),
+            insecure=dict(type='bool'),
+            packages=dict(type='str'),
+            update_packages=dict(type='bool'),
+            repo=dict(type='str'),
+            repo_gpg_key_url=dict(type='str', no_log=False),
```

Comment:
What does `no_log=False` do here? Isn't that default?

Reply:
It is, but Ansible has a check for parameters that *look* like secrets (e.g. it matches on "key"), and explicitly saying no log False here tells the check "yes, the param is a key but not a secret"